sunpy/net/vso/vso.py

```diff
@@ -777,6 +777,10 @@ def download_all(self, response, methods, downloader, path, qr, info=None):
             # If from_ and to are uninitialized, the else block of the loop
             # continues the outer loop and thus this code is never reached.
             # pylint: disable=W0631
+            if os.access(path, os.W_OK) is not True:
```

Comment:
Could you move this into [Fido](https://github.com/sunpy/sunpy/blob/master/sunpy/net/fido_factory.py#L299)